docs/guides/upstream-mtls.md

```diff
@@ -0,0 +1,158 @@
+---
+title: Upstream mTLS
+lang: en-US
+meta:
+  - name: keywords
+    content: pomerium identity-access-proxy mtls client-certificate
+description: >-
+  This guide covers how to configure Pomerium to provide mutual authentication
+  (mTLS) to an upstream service, using client certificates with a custom certificate authority.
+---
+
+# Upstream mTLS With Pomerium
+
+Part of a complete zero-trust security model is secure communication between your authenticated proxy (Pomerium) and the upstream service it provides access to. This means both Pomerium *and* the upstream service will authenticate each other.
+
+Pomerium confirms the identity of the upstream service by the TLS certificate it serves. See [`tls_custom_ca_file`] and [`tls_server_name`] for more information on configuring Pomerium to accept an upstream's TLS certificate.
+
+How an upstream service is configured to validate and accept client certificates is entirely up to that specific service and how it's configured. To provide a general explanation, in this guide we will use [OpenSSL] and [mkcert] to demonstrate how to configure Pomerium to provide the client certificate.
```

Comment:
```suggestion
Generally, every organization has a preferred method of managing internal certificates. How an upstream service is configured to validate and accept client certificates is entirely up to that specific service and how it's configured. To provide a general explanation, in this guide we will use [OpenSSL] and [mkcert] to demonstrate how to configure Pomerium to provide the client certificate.
```

Reply:
I think what we want to say here is:

- Managing internal TLS tends to be complicated (because you can't just use something like lets encrypt)
- Pomerium is flexibly enough to use whatever method of internal TLS the organization uses; whether that's using a service mesh, manual, or some sort of automated certificate management system